hudi-common/src/main/java/org/apache/hudi/common/table/log/AbstractHoodieLogRecordScanner.java

```diff
@@ -479,7 +481,7 @@ && compareTimestamps(logBlock.getLogBlockHeader().get(INSTANT_TIME), GREATER_THA
           continue;
         }
         if (logBlock.getBlockType() != COMMAND_BLOCK) {
-          if (this.tableVersion.lesserThan(HoodieTableVersion.EIGHT)) {
+          if (this.tableVersion.lesserThan(HoodieTableVersion.EIGHT) && !allowInflightInstants) {
```

Comment:
Could you fix `HoodieFileGroupReader`, `BaseHoodieLogRecordReader`, and `HoodieBaseFileGroupRecordBuffer` on the same logic?  `*LogRecordScanner` classes are going to be deprecated.

Reply:
@lokeshj1703 : can you follow up here.

Reply:
This may not be immediate necessity since we don't have support to reading MDT using new File group reader. But when @linliu-code gets to replacing it, we might need the support. So, we can take a call to delay the support for couple of weeks.

Reply:
@yihua This was required for RLI. RLI directly calls AbstractHoodieLogRecordScanner via `org.apache.hudi.metadata.HoodieTableMetadataUtil#getRevivedAndDeletedKeysFromMergedLogs`.